container_main.py

```
@@ -66,19 +66,34 @@ def run_scheduler():
-        os.environ["PORT"] = os.environ.get("PORT", "5000")
-
-        # Import and run the API server
-        from api_server import app
-
-        port = int(os.environ.get("PORT", 5000))
-        app.run(host="0.0.0.0", port=port, debug=False, use_reloader=False)
+        port_str = os.environ.get("PORT", "5000")
+
+        # Harden command invocation by strictly validating PORT as a digit (Sentinel 🛡️)
+        if not port_str.isdigit():
+            logger.error(f"Invalid PORT: {port_str}. Falling back to 5000.")
```

Ruff (0.15.1)

[warning] Logging statement uses f-string

(G004)